crates/cairo-lang-utils/src/bigint.rs

```diff
@@ -112,44 +112,39 @@ where
 
 impl Encode for BigIntAsHex {
     fn size_hint(&self) -> usize {
-        // sign, len, data.
-        1 + 8 + self.value.to_bytes_be().1.len()
+        // sign, data.
+        core::mem::size_of::<u8>() + self.value.bits() as usize / 8
     }
     fn encode_to<T: parity_scale_codec::Output + ?Sized>(&self, dest: &mut T) {
-        let (sign, data) = self.value.to_bytes_be();
-        match sign {
-            num_bigint::Sign::Minus => 0u8.encode_to(dest),
-            num_bigint::Sign::NoSign => 1u8.encode_to(dest),
-            num_bigint::Sign::Plus => 2u8.encode_to(dest),
-        };
-        // TODO(yair): better way to encode vec?
-        let len = data.len() as u64;
-        len.encode_to(dest);
-        for b in data.as_slice() {
-            b.encode_to(dest);
-        }
+        let (sign, data) = self.value.to_bytes_le();
+        (
+            match sign {
+                num_bigint::Sign::Minus => 0u8,
+                num_bigint::Sign::NoSign => 1u8,
+                num_bigint::Sign::Plus => 2u8,
+            },
+            Compact(data.len() as u64),
+        )
+            .encode_to(dest);
+        dest.write(&data);
     }
 }
 
 impl Decode for BigIntAsHex {
     fn decode<I: parity_scale_codec::Input>(
         input: &mut I,
     ) -> Result<Self, parity_scale_codec::Error> {
-        let discriminant = input.read_byte()?;
-        let sign = match discriminant {
+        let (sign, len) = <(u8, Compact<u64>)>::decode(input)?;
+        let sign = match sign {
             0u8 => num_bigint::Sign::Minus,
             1u8 => num_bigint::Sign::NoSign,
             2u8 => num_bigint::Sign::Plus,
             _ => {
                 return Err(parity_scale_codec::Error::from("Bad sign encoding."));
             }
         };
-        // TODO(yair): better way to decode vec?
-        let data_len = u64::decode(input)?;
-        let mut bytes: Vec<u8> = Vec::with_capacity(data_len as usize);
-        for _ in 0..data_len {
-            bytes.push(input.read_byte()?);
-        }
-        Ok(Self { value: BigInt::from_bytes_be(sign, bytes.as_slice()) })
+        let mut buffer = vec![0; len.0 as usize];
```

Comment:
I would rather use `Vec::with_capacity`, you don't have to initialize the memory to `0`. It will be overwritten just after without ever being read.